homeassistant/components/bosch_shc/switch.py

```diff
@@ -0,0 +1,232 @@
+
+    for switch in session.device_helper.smart_plugs:
+
+        entities.append(
+            SHCSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["smartplug"],
+            )
+        )
+
+    for switch in session.device_helper.light_switches:
+
+        entities.append(
+            SHCSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["lightswitch"],
+            )
+        )
+
+    for switch in session.device_helper.smart_plugs_compact:
+
+        entities.append(
+            SHCSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["smartplugcompact"],
+            )
+        )
+
+    for switch in session.device_helper.camera_eyes:
+
+        entities.append(
+            SHCSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["cameraeyes"],
+            )
+        )
+
+    for switch in session.device_helper.camera_360:
+
+        entities.append(
+            SHCSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["camera360"],
+            )
+        )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class SHCSwitch(SHCEntity, SwitchEntity):
+    """Representation of a SHC switch."""
+
+    entity_description: SHCSwitchEntityDescription
+
+    def __init__(
+        self,
+        device: SHCDevice,
+        parent_id: str,
+        entry_id: str,
+        description: SHCSwitchEntityDescription,
+    ) -> None:
+        """Initialize a SHC switch."""
+        super().__init__(device, parent_id, entry_id)
+        self.entity_description = description
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the switch."""
+        return (
+            getattr(self._device, self.entity_description.on_key)
+            == self.entity_description.on_value
+        )
+
+    @property
+    def today_energy_kwh(self) -> float | None:
+        """Return the total energy usage in kWh."""
+        if self.entity_description.has_consumption:
+            return self._device.energyconsumption / 1000.0
+        return None
+
+    @property
+    def current_power_w(self) -> float | None:
+        """Return the current power usage in W."""
+        if self.entity_description.has_consumption:
+            return self._device.powerconsumption
+        return None
+
```

Comment:
https://developers.home-assistant.io/docs/core/entity/switch?_highlight=switch#deprecated-properties

The following properties are deprecated and should not be used by new integrations. Provide them as sensors instead.

Reply:
Thanks for the hint, will do so.